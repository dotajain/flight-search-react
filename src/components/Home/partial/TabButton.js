import React from 'react';
import PropTypes from 'prop-types';

const TabButton = ({ ...props }) => (
  <button
    className={`tab__nav ${props.active ? 'tab__nav--active' : ''}`}
    type="button"
    onClick={props.onClick}
  >
    {props.label}
  </button>
);

TabButton.propTypes = {
  onClick: PropTypes.func,
  active: PropTypes.bool,
  label: PropTypes.string,
};
export default TabButton;
