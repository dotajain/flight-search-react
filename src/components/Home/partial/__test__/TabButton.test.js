import React from 'react';
import renderer from 'react-test-renderer';

import TabButton from '../TabButton';

global.fetch = jest.fn(
  () =>
    new Promise(resolve => {
      process.nextTick(() => resolve({ json: () => ({}) }));
    }),
);

it('Properly Rendered TabButton Component', () => {
  const component = renderer.create(<TabButton />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
