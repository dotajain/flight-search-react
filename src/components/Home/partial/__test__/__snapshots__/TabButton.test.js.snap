// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properly Rendered TabButton Component 1`] = `
<button
  className="tab__nav "
  onClick={undefined}
  type="button"
/>
`;
